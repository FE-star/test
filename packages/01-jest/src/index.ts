export function add(a: number, b: number): number {
  return a + b;
}

export function addFloat(a: number, b: number): number {
  const scale = 1000000;
  return (a * scale + b * scale) / scale;
}
