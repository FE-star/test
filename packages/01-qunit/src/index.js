function add(a, b) {
  return a + b;
}

function addFloat(a, b) {
  const scale = 1000000;
  return (a * scale + b * scale) / scale;
}

QUnit.test('test add()', function (assert) {
  assert.equal(add(1, 2), 3, 'add(1, 2) = 3');
});

QUnit.test('test addFloat()', function (assert) {
  assert.equal(addFloat(0.1, 0.2), 0.3, 'addFloat(0.1, 0.2) = 0.3');
});

// QUnit.test('test add() for float', function (assert) {
//   assert.equal(add(.1, .2), .3, 'add(.1, .2) = .3');
// });
