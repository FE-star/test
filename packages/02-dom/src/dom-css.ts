type DataItem = {
  id: number;
  name: string;
};

export function render() {
  const dataList: DataItem[] = [
    { id: 0, name: 'test name 0' },
    { id: 1, name: 'test name 1' },
    { id: 2, name: 'test name 2' },
    { id: 3, name: 'test name 3' },
    { id: 4, name: 'test name 4' }
  ];

  const root = document.querySelector('#root') as HTMLDivElement;

  const html = `
    <style>
    .app-container { background: rgb(240, 240, 240);  }
    .app-data-list { background: rgb(160, 160, 160);  }
    .app-data-item { background: rgb(192, 192, 192); color: rgb(34, 34, 34);  }
    </style>
    
    <div class="app-container">
      <div class="app-data-list">
        ${dataList
          .map((item) => {
            return `<div class="app-data-item">${item.name}</div>`;
          })
          .join('')}
      </div>
    </div>
  `;

  root.innerHTML = html;
}
