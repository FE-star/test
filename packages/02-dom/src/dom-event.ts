type DataItem = {
  id: number;
  name: string;
};

function renderList() {
  const dataList: DataItem[] = [
    { id: 0, name: 'test name 0' },
    { id: 1, name: 'test name 1' },
    { id: 2, name: 'test name 2' },
    { id: 3, name: 'test name 3' },
    { id: 4, name: 'test name 4' }
  ];

  const list = document.querySelector('#list') as HTMLDivElement;

  const html = ` 
    <div class="app-data-list">
      ${dataList
        .map((item) => {
          return `<div class="app-data-item">${item.name}</div>`;
        })
        .join('')}
    </div> 
  `;
  list.innerHTML = html;
}

function bindEvent() {
  const btn = document.querySelector('#btn-get-data');
  btn?.addEventListener('click', () => {
    renderList();
  });
}

export function renderAndBindEvent() {
  const root = document.querySelector('#root') as HTMLDivElement;
  root.innerHTML = `
  <div>
    <button id="btn-get-data">获取数据</button>
    <div id="list"></div>
  </div>
  `;
  bindEvent();
}
