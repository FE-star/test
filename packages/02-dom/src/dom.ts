type DataItem = {
  id: number;
  name: string;
};

export function render() {
  const dataList: DataItem[] = [
    { id: 0, name: 'test name 0' },
    { id: 1, name: 'test name 1' },
    { id: 2, name: 'test name 2' },
    { id: 3, name: 'test name 3' },
    { id: 4, name: 'test name 4' }
  ];

  const root = document.querySelector('#root') as HTMLDivElement;

  const html = `
    <div class="app-container">
      <div class="app-data-list">
        ${dataList
          .map((item) => {
            return `<div class="app-data-item">${item.name}</div>`;
          })
          .join('')}
      </div>
    </div>
  `;

  root.innerHTML = html;
}
