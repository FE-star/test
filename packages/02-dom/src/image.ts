export function renderImage() {
  const img = document.createElement('img');

  img.addEventListener('load', () => {
    console.log('load: 加载图片成功');
  });

  img.addEventListener('error', () => {
    console.log('error: 加载图片失败');
  });

  img.src = 'https://example.com/test.png';

  const root = document.querySelector('#root') as HTMLDivElement;

  root.appendChild(img);
}
