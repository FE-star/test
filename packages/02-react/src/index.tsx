import React, { useState } from 'react';

export function App() {
  const [count, setCount] = useState(0);

  return (
    <div className="app-counter">
      <div className="counter-text">count:{count}</div>
      <button
        className="counter-btn"
        onClick={() => {
          setCount(count + 1);
        }}
      >
        Add
      </button>
    </div>
  );
}
