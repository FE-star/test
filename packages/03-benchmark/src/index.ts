export function addFloat(a: number, b: number): number {
  const scale = 1000000;
  return (a * scale + b * scale) / scale;
}

export function addFloatBySplitStr(a: number, b: number): number {
  const aStrList = a.toString().split('.');
  const bStrList = b.toString().split('.');
  const resultInt = parseInt(aStrList[0]) + parseInt(bStrList[0]);
  const resultFloat = parseInt(aStrList[1]) + parseInt(bStrList[1]);
  return parseFloat(`${resultInt}.${resultFloat}`);
}
