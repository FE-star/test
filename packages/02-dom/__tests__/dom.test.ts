import { render } from '../src/dom';

describe('DOM', () => {
  beforeAll(() => {
    const root = document.createElement('div');
    root.setAttribute('id', 'root');
    document.body.appendChild(root);

    render();
  });

  test('snapshot', () => {
    const root = document.querySelector('#root');
    expect(root).toMatchSnapshot();
  });

  test('dom', () => {
    const root = document.querySelector('#root') as HTMLDivElement;
    const app = root.querySelector('.app-container') as HTMLDivElement;
    expect(app).toBeTruthy();

    const list = app.querySelector('.app-data-list') as HTMLDivElement;
    expect(list).toBeTruthy();

    const items = app.querySelectorAll('.app-data-item');
    expect(items.length).toBe(5);
  });
});
