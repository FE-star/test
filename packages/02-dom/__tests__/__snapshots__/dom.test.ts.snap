// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM snapshot 1`] = `
<div
  id="root"
>
  
    
  <div
    class="app-container"
  >
    
      
    <div
      class="app-data-list"
    >
      
        
      <div
        class="app-data-item"
      >
        test name 0
      </div>
      <div
        class="app-data-item"
      >
        test name 1
      </div>
      <div
        class="app-data-item"
      >
        test name 2
      </div>
      <div
        class="app-data-item"
      >
        test name 3
      </div>
      <div
        class="app-data-item"
      >
        test name 4
      </div>
      
      
    </div>
    
    
  </div>
  
  
</div>
`;
